src/python/pants/backend/adhoc/adhoc_tool.py

```diff
@@ -100,9 +104,46 @@ async def run_in_sandbox_request(
             target.address,
             target.get(AdhocToolExecutionDependenciesField).value,
             target.get(AdhocToolOutputDependenciesField).value,
+            target.get(AdhocToolRunnableDependenciesField).value,
         ),
     )
     dependencies_digest = execution_environment.digest
+    runnable_dependencies = execution_environment.runnable_dependencies
+
+    extra_env: dict[str, str] = dict(run_request.extra_env or {})
+    extra_path = extra_env.get("PATH", None)
+    if extra_path is not None:
+        del extra_env["PATH"]
+
+    extra_sandbox_contents = []
+
+    extra_sandbox_contents.append(
+        ExtraSandboxContents(
+            EMPTY_DIGEST,
+            extra_path,
+            FrozenDict(run_request.immutable_input_digests or {}),
+            FrozenDict(run_request.append_only_caches or {}),
+            FrozenDict(run_request.extra_env or {}),
```

Comment:
```suggestion
            run_request.immutable_input_digests or FrozenDict({}),
            run_request.append_only_caches or FrozenDict({}),
            run_request.extra_env or FrozenDict({}),
```

if mypy complains on this, maybe the typing on `ExtraSandboxContents` ought to adjust `FrozenDict` -> `Mapping`. Haven't checked if this would be the case.

Reply:
See https://github.com/pantsbuild/pants/pull/18347#discussion_r1120561782

Reply:
> if mypy complains on this, maybe the typing on `ExtraSandboxContents` ought to adjust `FrozenDict` -> `Mapping`. Haven't checked if this would be the case.

OK, but I still like my suggestion, but maybe the fix ought to go the other way for the run request type.


OR, we fix the FrozenDict invocation per DMs..

Reply:
Take a look at #99ddb301b4e3ee935affe9073d3da50b562678ae 

Now, we've got request types taking `FrozenDict` (for cacheability), and response types returning `Mapping`. We're using a new helper, `FrozenDict.frozen` to ensure that we're not needlessly creating duplicate `FrozenDicts`.